site/topics/logic/boolean-logic.rst

```diff
@@ -99,6 +99,155 @@ Properties of Logical Operators
+* De Morgan's laws are of particular interest in the context of computer architecture
+
+    * Given the popularity of **nor** and **nand** in circuit design
+
+
+* De Morgan's laws are as follows
+
+
+        :math:`\lnot a \lor \lnot b = \lnot(a \land b)`
+
+        :math:`\lnot a \land \lnot b = \lnot(a \lor b)`
+
+
+* Consider that the right hand side of both equations are **nand** and **nor** respectively
+* This means that, not only can **nand** be made with **not** and **and**, but also with **not** and **or**
+
+    * One does not even need the **and** operator to create **nand**
+    * In fact, given only **not** and **or**, one can create **and**
+
+        * :math:`\lnot(\lnot a \lor \lnot b) = \lnot(\lnot(a \land b)) = a \land b`
+
+
+    * This would work the same for **nor** being made with **not** and **and**
+
+
+* The below truth table provides an exhaustive proof of De Morgan's laws
+
+.. list-table:: De Morgan's Laws
+    :widths: auto
+    :header-rows: 1
+
+    * - :math:`a`
+      - :math:`b`
+      - :math:`\lnot a`
+      - :math:`\lnot b`
+      - :math:`a \lor b`
+      - :math:`a \land b`
+      - :math:`\lnot a \lor \lnot b`
+      - :math:`\lnot a \land \lnot b`
+      - :math:`\lnot (a \lor b)`
+      - :math:`\lnot (a \land b)`
+    * - ``0``
+      - ``0``
+      - ``1``
+      - ``1``
+      - ``0``
+      - ``0``
+      - ``1``
+      - ``1``
+      - ``1``
+      - ``1``
+    * - ``0``
+      - ``1``
+      - ``1``
+      - ``0``
+      - ``1``
+      - ``0``
+      - ``1``
+      - ``0``
+      - ``0``
+      - ``1``
+    * - ``1``
+      - ``0``
+      - ``0``
+      - ``1``
+      - ``1``
+      - ``0``
+      - ``1``
+      - ``0``
+      - ``0``
+      - ``1``
+    * - ``1``
+      - ``1``
+      - ``0``
+      - ``0``
+      - ``1``
+      - ``1``
+      - ``0``
+      - ``0``
+      - ``0``
+      - ``0``
+
+
+* De Morgan's laws may not be immediately obvious, but think about what they *mean*
+
+    * Visualizations may help with grasping the intuition
+
+
+* Consider the first law
+
+    :math:`\lnot a \lor \lnot b = \lnot(a \land b)`
+
+.. figure:: de_morgans_law_1.png
+    :width: 500 px
+    :align: center
+    :target: https://en.wikipedia.org/wiki/De_Morgan%27s_laws
+
+    Visualization of De Morgan's first law. In this figure, blue is what is included in the result, yellow is what is
+    excluded. Here, the union symbol (:math:`\cup`) is equivalent to **or** (:math:`\lor`) and the intersect symbol
+    (:math:`\cap`) is equivalent to **and** (:math:`\land`).
+
+
+* It is simpler to see how the above figure represents :math:`\lnot(a \land b)`
+
+    * Think of the Venn diagram of **and**, then invert it
+
+
+* But also imagine what :math:`\lnot a` and :math:`\lnot b` would be
+
+    * :math:`\lnot a` would be everything *but* what is within :math:`a`, including that which is in :math:`b`
+    * :math:`\lnot b` would be everything *but* what is within :math:`b`, including that which is in :math:`a`
```

Comment:
This took a few reads for me, maybe _...including that which is in both a and b_.

Reply:
That doesn't really work here as it's not a but b. 

Idk, I think it makes sense as is.

Reply:
To be clear, I mean:
> :math:`\lnot a` would be everything *but* what is within :math:`a`, including that which is in both :math:`a` and :math:`b`